<?xml version="1.0"?>
<launch>
	<!-- Arguments -->
	<arg name="in_tuning" default="False"/>
	<arg name ="setup_name" default="zed"/>
		<!-- Physical Parameters -->

	<!-- <include file="$(find-pkg-share common_params)/launch/common_params.launch"/> -->
	
	<node name="smoreo" pkg="smoreo" exec="smoreo" output="screen">
		<!-- Published topics -->
		<param name="/smoreo/predicted_landmarks" value="/perception/smoreo/detected"/>
		<param name="/smoreo/predicted_markers" value="/perception/smoreo/detected_markers"/>

		<!-- Subscribed topics -->
		<param name="/smoreo/bounding_boxes" value="/darknet_ros/bounding_boxes"/>
		<param name = "/smoreo/camera_info" value = "/zed/zed_node/left/camera_info" />
		<param name = "/smoreo/camera_info" value = "zed_left" />

		<!-- System params -->
		<!-- <param from="$(find-pkg-share smoreo)/config/$(var setup_name)Params.yaml" /> -->
		
		<!-- these are dummy numbers for testing only -->
		<param name="/smoreo/cut_off_y" value="1000" />
		<param name="/smoreo/cx" value="336.0" />
		<param name="/smoreo/cy" value="188.0" />
		<param name="/smoreo/f" value="336.0" />
		<param name = "/physical/cone_height" value = "0.4" />
		<param name = "/physical/camera_height_from_ground" value = "0.45"/>
		<param name = "/physical/cone_radius" value = "0.1"/>
		<param name = "/physical/lidar_height" value = "0.45"/>	
		<param name = "/smoreo/camera_frame" value = "zed_left" />
		<!-- - -->

		<param name="/smoreo/hardcode_params" value="True"/>
		<param name="/smoreo/use_cone_base" value= "True"/>
		<param name="/smoreo/in_tuning" value="$(var in_tuning)"/>



	</node>
	<!-- Nodes to be launched -->
</launch>